### PR TITLE
2.5 into develop

### PR DESCRIPTION
**Pull request by jameinel**
## Description of change

Another ratchet of changes in 2.5 that need to be in develop.
 * Merge pull request #9793 from jameinel/2.5-cleanup-volumes-1800872
 * Merge pull request #9788 from babbageclunk/raftlease-application-leaders
 * Merge pull request #9786 from howbazaar/2.5-move-worker
 * Merge pull request #9782 from jameinel/2.5-network-exit-1807615
 * Merge pull request #9781 from timClicks/2.5
 * Merge pull request #9778 from anastasiamac/current-controller-saga
 * Merge pull request #9776 from SimonRichardson/lxd-sub-acceptance-test
 * Merge pull request #9777 from anastasiamac/storage-err-fix-machine-removal-25
 * Merge pull request #9759 from wallyworld/k8s-ephemeral-storage
 * Merge pull request #9773 from achilleasa/emit-errors-for-invalid-credential-names
 * Merge pull request #9774 from achilleasa/backport-leader-unit-support-for-juju-run
 * Merge pull request #9764 from manadart/2.5-maas-container-inherit-props
 * Merge pull request #9761 from Veebers/update-cloud-controller
 * Merge pull request #9770 from hmlanigan/2.5-lxd-profile

## QA steps

See individual patches.

## Documentation changes

See individual patches.

## Bug reference

 * https://bugs.launchpad.net/juju/+bug/1800872
 * https://bugs.launchpad.net/juju/+bug/1807615
 * https://bugs.launchpad.net/juju/+bug/1786140
 * https://bugs.launchpad.net/juju/+bug/1813079
 * https://bugs.launchpad.net/juju/+bug/1702538
 * https://bugs.launchpad.net/juju/+bug/1816826
 * https://bugs.launchpad.net/juju/+bug/1815636

### CODE REVIEW

**Comment by jameinel** [comment]
$$merge$$ self approving this as it is a merge of 2.5 into develop

File: cmd/juju/cloud/addcredential.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/juju/juju/cmd/juju/interact"
 	"github.com/juju/juju/environs"
 	"github.com/juju/juju/jujuclient"
+	"gopkg.in/juju/names.v2"
```
Comment (anastasiamac) [inline cmd/juju/cloud/addcredential.go]: This is in the wrong import clause.

File: cmd/modelcmd/controller.go [hunk 2]
```diff
@@ -104,6 +104,12 @@ func (c *ControllerCommandBase) initController() error {
 }
 
 // DetermineCurrentController returns current controller on this client.
+// The returned result corresponds to, in order of precedence:
```
Comment (anastasiamac) [inline cmd/modelcmd/controller.go]: This has been removed from all 2.5 and develop versions.

Comment (jameinel) [reply cmd/modelcmd/controller.go]: thanks for the heads up, will fix in the next merge if this lands as is.

File: cmd/juju/cloud/addcredential.go (hunk 1 above)
Comment (jameinel) [reply cmd/juju/cloud/addcredential.go]: thanks

File: cmd/modelcmd/controller.go (hunk 2 above)
Comment (jameinel) [reply cmd/modelcmd/controller.go]: let me know if https://github.com/juju/juju/pull/9821 doesn't address your concerns.

**Pull request merged**